server/py/services/alerts/crud/alerts.py

```diff
@@ -403,31 +352,66 @@ def _event_entity_matches(alert_entity, event_entity):
+    def _validate_alert(
+        self,
+        alert: mlrun.common.schemas.AlertConfig,
+        name: str,
+        project: str,
+    ):
         self.validate_alert_name(alert.name)
         if name != alert.name:
             raise mlrun.errors.MLRunBadRequestError(
                 f"Alert name mismatch for alert {name} for project {project}. Provided {alert.name}"
             )
 
-        if alert.criteria is not None:
-            if alert.criteria.count >= mlconfig.alerts.max_criteria_count:
+        self._validate_alert_criteria(project, name, alert.criteria)
+        self._validate_alert_notifications(project, name, alert.notifications)
+
+        if alert.entities.project != project:
+            raise mlrun.errors.MLRunBadRequestError(
+                f"Invalid alert entity project ({alert.entities.project}) for alert {name} for project {project}"
+            )
+
+    @staticmethod
+    def _validate_alert_criteria(
+        project: str,
+        name: str,
+        criteria: mlrun.common.schemas.AlertCriteria,
+    ):
+        """
+        Validate the alert criteria, ensuring:
+        - The number of criteria does not exceed the maximum allowed.
+        - If a period is specified, it is a valid duration.
+        """
+        if criteria is not None:
+            if criteria.count >= mlconfig.alerts.max_criteria_count:
                 raise mlrun.errors.MLRunPreconditionFailedError(
-                    f"Maximum criteria count exceeded: {alert.criteria.count}"
+                    f"Maximum criteria count exceeded: {criteria.count}"
                 )
-
             if (
-                alert.criteria.period is not None
+                criteria.period is not None
                 and framework.utils.helpers.string_to_timedelta(
-                    alert.criteria.period, raise_on_error=False
+                    criteria.period, raise_on_error=False
                 )
                 is None
             ):
                 raise mlrun.errors.MLRunBadRequestError(
-                    f"Invalid period ({alert.criteria.period}) specified for alert {name} for project {project}"
+                    f"Invalid period ({criteria.period}) specified for alert {name} for project {project}"
                 )
 
-        for alert_notification in alert.notifications:
+    @staticmethod
+    def _validate_alert_notifications(
+        project: str,
+        name: str,
+        notifications: list[mlrun.common.schemas.AlertNotification],
+    ):
+        """
+        Validate the notifications configured for the alert, ensuring:
+        - Each notification's kind is supported (git, slack, webhook).
+        - Each notification's structure adheres to the defined notification schema.
+        - If a cooldown period is specified, it must be a valid time string.
+        """
+        for alert_notification in notifications:
             if alert_notification.notification.kind not in [
```

Comment:
We can have a method in the notification schema class for `ValidAlertNotificationKinds`